### PR TITLE
feat: use `bun serve` in http tests

### PR DESCRIPTION
**Pull request by barbmarcio**
Drops a dependency and simplifies serving assets related to e2e tests.

### CODE REVIEW

File: e2e/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { file } from "bun";
+
+const PORT = process.env.SERVER_PORT || 8080;
+
+Bun.serve({
+  fetch(req) {
+    const url = new URL(req.url);
+    const pathname = url.pathname === "/" ? "/index.html" : url.pathname;
+    const filePath = `./dist${pathname}`;
```
Comment (jbergstroem) [inline e2e/index.ts:9]: Shouldn't this be:
```suggestion
    const filePath = `./${pathname}`;
```

File: package.json [hunk 2]
```diff
@@ -36,13 +35,13 @@
     "test:e2e": "playwright test",
     "format": "biome check",
     "format:fix": "biome check --write",
-    "prepare": "lefthook install"
+    "prepare": "lefthook install",
+    "serve": "bun run ./e2e/index.ts"
```
Comment (jbergstroem) [inline package.json:39]: Lets indicate that this is e2e related
```suggestion
    "serve:e2e": "bun run ./e2e/index.ts"
```

File: playwright.config.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export default defineConfig({
     },
   ],
   webServer: {
-    command: `http-server ./ -p ${process.env.PLAYWRIGHT_PORT || 8080}`,
+    command: "bun run serve",
```
Comment (jbergstroem) [inline playwright.config.ts:26]:
```suggestion
    command: "bun run serve:e2e",
```

File: tsconfig.json [hunk 4]
```diff
@@ -17,6 +17,6 @@
     "allowJs": true,
     "types": ["node"]
   },
-  "include": ["src/*.ts", "src/**/*.ts"],
+  "include": ["src/*.ts", "src/**/*.ts", "e2e/server.ts"],
```
Comment (jbergstroem) [inline tsconfig.json:20]: I actually prefer `server.ts` over `index.ts` but you seem to use `index.ts` elsewhere:
```suggestion
  "include": ["src/*.ts", "src/**/*.ts", "e2e/index.ts"],
```

**Review by jbergstroem — CHANGES_REQUESTED** [review]
Do we need the favicon?

File: e2e/index.ts (hunk 1 above)
Comment (barbmarcio) [reply e2e/index.ts:9]: As we are serving the `dist`, if we remove it from the template string, it would consider we are trying to serve the root folder

Comment (jbergstroem) [reply e2e/index.ts:9]: Oh, I missed that you removed the file router. Never mind.

**Comment by jbergstroem** [comment]
Question: why did you increase the retry? If this fails it'd be interesting to know why.

**Comment by barbmarcio** [comment]
> Question: why did you increase the retry? If this fails it'd be interesting to know why.

Actually by default was set to 1. In order to test and not wait for the retries I've changed to 0, now just changing back to 1 :)

**Comment by barbmarcio** [comment]
> LGTM but I'd stick with 0 retries and pick up any failures as they arrive.

Agreed 100%, just changed it back

**Pull request merged**